langchain4j-ovh-ai/src/test/java/dev/langchain4j/model/ovhai/OvhAiEmbeddingModelIT.java

```diff
@@ -0,0 +1,56 @@
+package dev.langchain4j.model.ovhai;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import org.junit.jupiter.api.Test;
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.model.embedding.EmbeddingModel;
+import dev.langchain4j.model.output.Response;
+
+public class OvhAiEmbeddingModelIT {
+
+    
+
+    @Test
+    void should_embed() {
+        EmbeddingModel model = OvhAiEmbeddingModel
+        .builder()
+        .apiKey(System.getenv("OVHAI_AI_API_KEY"))
+        .logRequests(true)
+        .logResponses(true)
+        .build();
+
+        // given
+        TextSegment textSegment = TextSegment.from("Embed this sentence.");
+
+        // when
+        Response<Embedding> response = model.embed(textSegment);
+
+        // then
+        assertThat(response.content().vector()).hasSize(768);
+
+        assertThat(response.finishReason()).isNull();
+    }
+
+    @Test
+    void should_embed_with_bge_e5_model() {
+        EmbeddingModel model = OvhAiEmbeddingModel
+        .builder()
+        .apiKey(System.getenv("OVHAI_AI_API_KEY"))
+        .baseUrl("https://bge-base-en-v1-5.endpoints.kepler.ai.cloud.ovh.net")
+        .logRequests(true)
+        .logResponses(true)
+        .build();
+
+        // given
+        TextSegment textSegment = TextSegment.from("Embed this sentence.");
+
+        // when
+        Response<Embedding> response = model.embed(textSegment);
+
+        // then
+        assertThat(response.content().vector()).hasSize(768);
+
+        assertThat(response.finishReason()).isNull();
+    }
+}
```

Comment:
please add a test that embeds multiple segments at once

Reply:
I have one problem: our embedding endpoints don't support the multiple segments at once 😭.

I have three options:

1. This is a limitation and I throw an exception if there are more than one segment
2. I'll develop the multiple call in the client (not very good in performance but allows the presence of the multiple segments feature)
3. I'll wait for my developers develop this feature in our endpoints and after I update the PR

I can do 2 then open a new PR with 3 if you want.
Or we wait 3 if you prefer, tell me 😉 .

Reply:
"2 now, 3 later" sounds good :)

Reply:
@langchain4j the commit https://github.com/langchain4j/langchain4j/pull/1355/commits/771037f2ac79b3aa50e54ac6b847df2bbb6ee044 is a first version to a _embedAll_ solution with its corresponding test.

Not sure it's the more efficient but it is a first implementation 😉